packages/suite/src/actions/wallet/stake/stakeFormSolanaActions.ts

```diff
@@ -0,0 +1,244 @@
+import { LAMPORTS_PER_SOL } from '@solana/web3.js';
+
+import { BigNumber } from '@trezor/utils/src/bigNumber';
+import TrezorConnect, { FeeLevel } from '@trezor/connect';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import {
+    calculateTotal,
+    calculateMax,
+    getExternalComposeOutput,
+    formatAmount,
+    getPubKeyFromAddress,
+} from '@suite-common/wallet-utils';
+import {
+    StakeFormState,
+    PrecomposedLevels,
+    PrecomposedTransaction,
+    PrecomposedTransactionFinal,
+    ExternalOutput,
+    AddressDisplayOptions,
+} from '@suite-common/wallet-types';
+import { selectDevice, ComposeActionContext } from '@suite-common/wallet-core';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import {
+    MIN_SOL_AMOUNT_FOR_STAKING,
+    MIN_SOL_BALANCE_FOR_STAKING,
+    MIN_SOL_FOR_WITHDRAWALS,
+} from '@suite-common/wallet-constants';
+
+import { Dispatch, GetState } from 'src/types/suite';
+import { selectAddressDisplayType } from 'src/reducers/suite/suiteReducer';
+import { prepareStakeSolTx } from 'src/utils/suite/solanaStaking';
+
+
+
+const calculate = (
```

Comment:
au, this file is really big copy-paste. can we try to avoid that, please

Reply:
yes, I know. Refactored this file here https://github.com/trezor/trezor-suite/pull/15409/commits/af37cda7bcb7ef12ab440b18d6f0dd32f93f5a38
I think signTransaction functions are pretty unique to keep them as is for now.